### ISSUE TITLE
Rabbitmq broker can't handle infinite time limit of a task

### ISSUE THREAD

**Issue by jan-zmeskal** [celery/celery#7708]
# Checklist

- [x] I have verified that the issue exists against the `master` branch of Celery.
- [ ] This has already been asked to the [discussions forum](https://github.com/celery/celery/discussions) first.
- [x] I have read the relevant section in the
  [contribution guide](https://docs.celeryq.dev/en/master/contributing.html#other-bugs)
  on reporting bugs.
- [x] I have checked the [issues list](https://github.com/celery/celery/issues?q=is%3Aissue+label%3A%22Issue+Type%3A+Bug+Report%22+-label%3A%22Category%3A+Documentation%22)
  for similar or identical bug reports.
- [x] I have checked the [pull requests list](https://github.com/celery/celery/pulls?q=is%3Apr+label%3A%22PR+Type%3A+Bugfix%22+-label%3A%22Category%3A+Documentation%22)
  for existing proposed fixes.
- [x] I have checked the [commit log](https://github.com/celery/celery/commits/master)
  to find out if the bug was already fixed in the master branch.
- [x] I have included all related issues and possible duplicate issues
  in this issue (If there are none, check this box anyway).

## Mandatory Debugging Information

- [x] I have included the output of ``celery -A proj report`` in the issue.
    (if you are not able to do this, then at least specify the Celery
     version affected).
- [x] I have verified that the issue exists against the `master` branch of Celery.
- [x] I have included the contents of ``pip freeze`` in the issue.
- [x] I have included all the versions of all the external dependencies required
  to reproduce this bug.

## Optional Debugging Information

- [x] I have tried reproducing the issue on more than one Python version
  and/or implementation.
- [x] I have tried reproducing the issue on more than one message broker and/or
  result backend.
- [x] I have tried reproducing the issue on more than one version of the message
  broker and/or result backend.
- [ ] I have tried reproducing the issue on more than one operating system.
- [ ] I have tried reproducing the issue on more than one workers pool.
- [ ] I have tried reproducing the issue with autoscaling, retries,
  ETA/Countdown & rate limits disabled.
- [ ] I have tried reproducing the issue after downgrading
  and/or upgrading Celery and its dependencies.

#### Related Issues

- None

#### Possible Duplicates

- None

## Environment & Settings

**Celery version**: 5.3.0b1 (dawn-chorus)

<details>
<summary><b><code>celery report</code> Output:</b></summary>
<p>

```
software -> celery:5.3.0b1 (dawn-chorus) kombu:5.3.0b1 py:3.8.10
            billiard:4.0.2 py-amqp:5.1.1
platform -> system:Linux arch:64bit, ELF
            kernel version:5.15.0-43-generic imp:CPython
loader   -> celery.loaders.app.AppLoader
settings -> transport:amqp results:disabled

broker_url: 'amqp://guest:********@localhost:5672//'
deprecated_settings: None

```

</p>
</details>

# Steps to Reproduce

## Required Dependencies

* **Minimal Python Version**: Python 3.8.10
* **Minimal Celery Version**: celery:5.3.0b1
* **Minimal Kombu Version**: kombu 5.3.0b1
* **Minimal Broker Version**: rabbitmq 3.10.7
* **Minimal Result Backend Version**: N/A or Unknown
* **Minimal OS and/or Kernel Version**: 5.15.0-43-generic
* **Minimal Broker Client Version**: amqp 5.1.1
* **Minimal Result Backend Client Version**: N/A or Unknown

### Python Packages

<details>
<summary><b><code>pip freeze</code> Output:</b></summary>
<p>

```
amqp==5.1.1
billiard==4.0.2
celery==5.3.0b1
click==8.1.3
click-didyoumean==0.3.0
click-plugins==1.1.1
click-repl==0.2.0
kombu==5.3.0b1
prompt-toolkit==3.0.30
pytz==2022.2
six==1.16.0
typing-extensions==4.3.0
vine==5.0.0
wcwidth==0.2.5
```

</p>
</details>

### Other Dependencies

<details>
<p>
docker-compose file I use to start rabbitmq broker:

```
version: "3.9"

services:
  rabbitmq:
    image: rabbitmq:3.10.7-management
    ports:
      - "5672:5672"
      - "15672:15672"
```
</p>
</details>

## Minimally Reproducible Test Case

<details>
<p>

```python
from celery import Celery

app = Celery('tasks', broker='amqp://guest:guest@localhost:5672')

@app.task(time_limit=float('inf'))
def add(x, y):
    return x + y


if __name__ == '__main__':
    add.delay(3, 4)
```

</p>
</details>

# Expected Behavior

When executing the provided python module, I'd expect the task to be passed to a worker. The worker should of course report correct result.

If setting infinite time limit is prohibited, it should be at the very least mentioned in the docs. Better yet, I'd expect worker to error out when registering such task.

# Actual Behavior

We ran into this issue in much more complicated application, but I found out that it's actually very easy to reproduce with couple lines of code. As far as I can tell from celery docs, there's no upper limit on task's time limits. We have one task and we need it to have an infinite time limit. I know this isn't the best practice, but I believe we have a good reason for it and though it through. Anyway, celery doesn't prevent you from doing it in any way.

So the question is: how do you set an infinite time limit? We found out that you cannot use `@app.task(time_limit=None)`, because if there is application-wide default, it will be applied on task decorated in such way. The only solution we could come up with is this:
```python
@app.task(time_limit=float("inf")
def my_task():
    ...
```
This is perfectly fine from python point of view and celery worker does not complain when it registers such task:
<details>
<summary>Worker output</summary>
<p>

```
 -------------- celery@t14 v5.3.0b1 (dawn-chorus)
--- ***** ----- 
-- ******* ---- Linux-5.15.0-43-generic-x86_64-with-glibc2.29 2022-08-12 15:34:43
- *** --- * --- 
- ** ---------- [config]
- ** ---------- .> app:         tasks:0x7ff9d0b14d30
- ** ---------- .> transport:   amqp://guest:**@localhost:5672//
- ** ---------- .> results:     disabled://
- *** --- * --- .> concurrency: 16 (prefork)
-- ******* ---- .> task events: OFF (enable -E to monitor tasks in this worker)
--- ***** ----- 
 -------------- [queues]
                .> celery           exchange=celery(direct) key=celery
                

[tasks]
  . tasks.add

[2022-08-12 15:34:43,469: WARNING/MainProcess] /home/jan/repo/celery-reproduction/.venv/lib/python3.8/site-packages/celery/worker/consumer/consumer.py:491: CPendingDeprecationWarning: The broker_connection_retry configuration setting will no longer determine
whether broker connection retries are made during startup in Celery 6.0 and above.
If you wish to retain the existing behavior for retrying connections on startup,
you should set broker_connection_retry_on_startup to True.
  warnings.warn(

[2022-08-12 15:34:43,476: INFO/MainProcess] Connected to amqp://guest:**@127.0.0.1:5672//
[2022-08-12 15:34:43,477: WARNING/MainProcess] /home/jan/repo/celery-reproduction/.venv/lib/python3.8/site-packages/celery/worker/consumer/consumer.py:491: CPendingDeprecationWarning: The broker_connection_retry configuration setting will no longer determine
whether broker connection retries are made during startup in Celery 6.0 and above.
If you wish to retain the existing behavior for retrying connections on startup,
you should set broker_connection_retry_on_startup to True.
  warnings.warn(

[2022-08-12 15:34:43,496: INFO/MainProcess] mingle: searching for neighbors
[2022-08-12 15:34:44,532: INFO/MainProcess] mingle: all alone
[2022-08-12 15:34:44,565: INFO/MainProcess] celery@t14 ready.
```

</p>
</details>

However, when the python code above is executed, and `add.delay(3,4)` is called, following error is found in rabbitmq broker and the task isn't received by the worker. See the log here: https://gist.github.com/jan-zmeskal/8fa55027f6c579373cba8ff2c15ab26f

I recreated the same scenario while using `redis:7.0.4` container as broker and the task was successfully processed by the worker. Could you please look into this? Changing our broker backend could be quite problematic for us. Thank you!

**Comment by auvipy** [comment]
my question, can you use a bigger time limit value and get your desired work done?

**Comment by jan-zmeskal** [comment]
@auvipy Hi. Yes, that's the way I decided to work around the issue. As far as we are concerned, we can live with that. I just wanted to point this issue out because others might hit it as well and it was quite hard to debug it.

**Comment by auvipy** [comment]
yes, thanks for that. we are going to keep this open for future fix

**Comment by Zigars** [comment]
Hi, @auvipy @jan-zmeskal 
I have this problem too. my celery version is 5.3.0, and used redis as the broker and backend.

task state was tend to "PENDING" in **24h**. although I haven't set any time_limit, I didn't found any error in my task, and the tasks are still running in the background. but my state api tell me the task is failed("PENDING").

I'm very confusing about it. I need the task to run continuously in the background and the task status to remain started until encountering an error stop. I'll try set time_limit=float("inf") to solve the problem.

![image](https://github.com/celery/celery/assets/32835472/2083aee7-b431-40f1-be2b-9236a9f5f42c)
![image](https://github.com/celery/celery/assets/32835472/fa29964a-639a-44cb-bb16-b3d3e62c6a6d)
